tests/unit/base/SGObject_unittest.cc

```diff
@@ -99,6 +99,74 @@ TEST(SGObject, equals_other_has_NULL_parameter)
 	SG_UNREF(kernel2);
 }
 
+TEST(SGObject, clone_not_null_and_correct_type)
+{
+	auto object = some<CGaussianKernel>();
+	auto clone = object->clone();
+	ASSERT_NE(clone, nullptr);
+	ASSERT_TRUE(!strcmp(object->get_name(), clone->get_name()));
+}
+
+TEST(SGObject, clone_basic_parameter)
+{
+	auto object = some<CGaussianKernel>();
+	object->put("log_width", 2.0);
+	ASSERT_EQ(object->get<float64_t>("log_width"), 2);
+
+	auto clone = object->clone();
+	EXPECT_EQ(
+	    object->get<float64_t>("log_width"),
+	    clone->get<float64_t>("log_width"));
+}
+
+TEST(SGObject, clone_sgmatrix_parameter)
+{
+	SGMatrix<float64_t> data(10, 10);
+	for (auto i : range(data.num_rows * data.num_cols))
+		data.matrix[i] = CMath::randn_double();
+
+	auto object = some<CDenseFeatures<float64_t>>(data);
+	ASSERT_TRUE(data.equals(object->get_feature_matrix()));
+
+	auto clone = object->clone();
+	EXPECT_NE(
```

Comment:
@lisitsyn this is what I meant in my other comment. This assertion here fails, because the pointer is the same

Reply:
ok got it now. Seems trivial to add? :)

Reply:
I'll do it :)

Reply:
#4083